### PR TITLE
Allow order with multiple line items to be marked as "Returned"

### PR DESCRIPTION
**Pull request by spaghetticode**
**Description**

This PR fixes issue #3198 

When a customer return is created for an order with multiple line items and the
return state of all the items is `Received` then the order state should change
to `Returned` just like it happens with orders with only one line item.

**Checklist:**
- [x] I have followed [Pull Request guidelines](https://github.com/solidusio/solidus/blob/master/CONTRIBUTING.md#pull-request-guidelines)
- [x] I have added a detailed description into each commit message
- [x] I have added tests to cover this change (if needed)

### CODE REVIEW

File: core/app/models/spree/order.rb [hunk 1]
```diff
@@ -286,7 +286,7 @@ def allow_cancel?
     end
 
     def all_inventory_units_returned?
-      inventory_units.all?(&:returned?)
+      inventory_units.returned.count == inventory_units.count
```
[the remark this answers is not in the document]
Comment (spaghetticode) [reply core/app/models/spree/order.rb]: This code is executed for each return item, for a better understanding of the process you can start from `ReturnItem#process_inventory_unit!`. The first inventory unit is marked as `returned`, and eventually `order.inventory.units` is called. 

The query is executed only for the first return item, and as we know there's no issue as long as the customer return is for one single item. But when there's more than one, from the second time on, `order.inventory.units` refers to stale data, and must be updated. Using `count` ensures a new (lightweight) query is performed. 

Basically, it happens something similar to this:
```ruby
o = create :shipped_order, line_items_count: 2
o.inventory_units.all? &:shipped?
=> true

u = Spree::InventoryUnit.find(o.inventory_units.first.id)
u.return!
u.reload.returned?

o.inventory_units.all? &:shipped?
=> true
```

File: core/app/models/spree/return_item.rb [hunk 2]
```diff
@@ -233,10 +235,12 @@ def currency
 
     def process_inventory_unit!
       inventory_unit.return!
-
       if customer_return
         customer_return.stock_location.restock(inventory_unit.variant, 1, customer_return) if should_restock?
-        customer_return.process_return!
+        unless skip_customer_return_processing
```
Comment (kennyadsl) [inline core/app/models/spree/return_item.rb]: I think we should also test that a deprecation warning is raised in some circumstances.

File: backend/spec/controllers/spree/admin/reimbursements_controller_spec.rb [hunk 3]
```diff
@@ -38,7 +38,10 @@
     let(:order) { customer_return.order }
     let(:return_item) { customer_return.return_items.first }
     let(:payment) { order.payments.first }
-    before { return_item.receive! }
+    before do
+      return_item.skip_customer_return_processing = true
```
Comment (kennyadsl) [inline backend/spec/controllers/spree/admin/reimbursements_controller_spec.rb]: Do you think it could be a good idea to add this as default in the return item factory?

File: core/app/models/spree/return_item.rb (hunk 2 above)
Comment (spaghetticode) [reply core/app/models/spree/return_item.rb]: Makes sense, I added this spec: https://github.com/solidusio/solidus/pull/3199/files#diff-0f55d3185362ecf4697cad5429703bd0

File: backend/spec/controllers/spree/admin/reimbursements_controller_spec.rb (hunk 3 above)
Comment (spaghetticode) [reply backend/spec/controllers/spree/admin/reimbursements_controller_spec.rb]: I totally agree 👍 
This also reduced the changes I had to make to spec files
